### ISSUE TITLE
Consider adding `filterSet` to `IsSet`

### ISSUE THREAD

**Issue by lehmacdj** [snoyberg/mono-traversable#177]
Just like `filterMap` which was added in #167, we are missing
```
filterSet :: IsSet set => (Element set -> Bool) -> set -> set
```
Can we add such a function?

In principle it is possible to give an implementation of this for every instance of `SetContainer` but we wouldn't be able to give a default implementation for it, because we don't have `setFromList`.

**Comment by snoyberg** [comment]
I'm open to such a PR.

**Comment by lehmacdj** [comment]
Implemented in #178
